fix: upgrade level libs to resolve node 10 failure

jacobheun opened the pull request:
This upgrades the versions of leveldown, levelup and level-js to resolve a node 10.x build issue in ipfs-repo.

This also moves the responsibility of shiming leveldown with level-js for the browser. The version of level-js has been changed to match the current version in js-ipfs-repo.

I've run this against the ipfs-repo test suite, but still would like to run it against js-ipfs.

connects to ipfs/js-ipfs#1347

achingbrain commented:
It would be great if this could get merged as I'm seeing Level/leveldown#194 in js-ipfs which seems to have been resolved but the fix was released in a later version.

alanshaw commented on src/index.js:
```diff
@@ -24,10 +24,24 @@ class LevelDatastore {
-    }))
+    // Default to leveldown db
+    const database = opts && opts.db ? opts.db : require('leveldown')
+    delete opts.db
```
needs an existence check like the line above otherwise `ReferenceError: opts is not defined` for `null`/`undefined`

alanshaw replied:
...and a test? ;)

jacobheun commented:
@alanshaw yes, thanks for pointing that out. I've added tests for default and override configuration. I've also added levedown to the interface test suites so we can ensure those pluggable databases are working properly with dependency upgrades.

jacobheun commented:
@alanshaw this should be good, can you take a look and approve if all looks good?  This is being used in the js-ipfs node 10 fix which I have jenkins builds passing for https://github.com/ipfs/js-ipfs/pull/1358/files.

jacobheun commented:
@dignifiedquire thanks for pointing that out, I updated the readme to reference the shimming expectations. This should be good to merge now, once that's verified and it's re-approved.

vweevers commented on src/index.js:
```diff
@@ -24,10 +24,29 @@ class LevelDatastore {
+      database(path),
+      Object.assign({}, opts, {
+        compression: false, // same default as go
+        valueEncoding: 'binary'
```
FYI `levelup` v2 no longer includes encodings, so this `valueEncoding` option will have no effect.

Instead, wrap `database(path)` with `encoding-down`. See [UPGRADING.md](https://github.com/level/levelup/blob/HEAD/UPGRADING.md#upgrade-guide-1) for more.

jacobheun commented:
Thanks for pointing that out @vweevers, I've updated the code to reflect the levelup upgrading guide and tested this against ipfs-repo. Pending reviews, this should good to go.
